### PR TITLE
allow hostNetwork, hostPID and hostIPC by default

### PR DESCRIPTION
**Pull request by ArtfulCoder**
@bgrant0607

### CODE REVIEW

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -173,9 +173,9 @@ func NewKubeletServer() *KubeletServer {
 		FileCheckFrequency:          20 * time.Second,
 		HealthzBindAddress:          net.ParseIP("127.0.0.1"),
 		HealthzPort:                 10248,
-		HostNetworkSources:          kubelet.FileSource,
-		HostPIDSources:              kubelet.FileSource,
-		HostIPCSources:              kubelet.FileSource,
+		HostNetworkSources:          kubelet.AllSource,
```
Comment (bgrant0607) [inline cmd/kubelet/app/server.go]: Please update the flag documentation also:
https://github.com/kubernetes/kubernetes/blob/7f9df8c062a693fb05923b1407d5995724f96b44/cmd/kubelet/app/server.go#L231

**Comment by bgrant0607** [comment]
Please file an issue for this and also cherrypick into 1.1 after merge.

**Comment by ArtfulCoder** [comment]
Addresses issue #15031

File: cmd/kubelet/app/server.go (hunk 1 above)
Comment (ArtfulCoder) [inline cmd/kubelet/app/server.go]: done

**Comment by ArtfulCoder** [comment]
PTAL

**Comment by bgrant0607** [comment]
LGTM

**Comment by sttts** [comment]
:+1:

**Comment by ArtfulCoder** [comment]
Release notes: Users can now launch pods with containers that have hostNetwork enabled. Before 1.1 release, such pods with be in pending state and would never run.

**Pull request merged**